plugins/filter.rs

```diff
@@ -0,0 +1,9 @@
+use std::collections::HashMap;
+use crate::rpc;
+
+pub type CodeletType = fn(&Filter, &rpc::Rpc) -> Option<rpc::Rpc>;
```

Comment:
There is a circular dependency here:

1. CodeletType is a function type that takes an argument of type Filter.
2. Filter is a struct that includes a CodeletType field.

Can we avoid this circular dependency (much more preferable) or at least add comments on why such a circular dependency is needed?

Reply:
The filter no longer references the CodeletType;  I made it have only one field, and it always has to have an implementation with a function named execute.  This should be a bit logically clearer.